packages/builtin_test/_test.pony

```diff
@@ -306,6 +306,27 @@ class \nodoc\ iso _TestStringToFloat is UnitTest
       h.assert_true("+INFINITY".f64()?.infinite())
     }, "+INFINITY")
 
+    // Test that non-null terminated strings are well behaved
+    //
+    // We craft a string that contains 1.1\0, chop the underlying
+    // array, and then create a new string that spans only the
+    // first character.
+    //
+    // The implementation of f32 / f64 errors if:
+    // 1. strof/strod fail by setting errno, or
+    // 2. strof/strod don't consume the entire string
+    //
+    // Point (2) is supposed to cover things like "1.23abc",
+    // but it also covers the situations where strod/strof
+    // walk past the end of our memory allocation.
+    //
+    // As such, a buggy version that calls strof/strod without
+    // checking for a null terminator will raise an error due to (2),
+    // since C will happily walk the entire buffer until it reaches
+    // \0 at the end of str2.
+    (let str1: String val, let str2) = "1.1".clone().chop(1)
```

Comment:
Using `trim` here would be simpler and probably easier to follow.
```suggestion
    let str1 = "1.1"trim(0, 1)
```

We don't need `clone` or `chop` because we don't need `iso` to make this test work.

Reply:
Okay, I've changed it. I liked the `chop` example because it was easy to point to `str2` when saying that C would happily read from that memory. I've reworded a bit the text above it to match the new example.